src/site/content/en/angular/virtualize-lists-with-cdk/index.md

```diff
@@ -0,0 +1,118 @@
+
+```
+npm install --save @angular/cdk
+```
+
+### Add `ScrollingModule` to your app
+With the CDK installed, import `ScrollingModule`, which handles virtual scrolling, from the `@angular/cdk/scrolling` package. Then add it to the imports array of your module:
+
+```js
+import {ScrollingModule} from '@angular/cdk/scrolling';
+
+...
+imports: [
+  ScrollingModule
+...
+]
+...
+```
+
+### Create a viewport
+To see how the package works, try creating a component with a simple list of numbers from 0 to 99,999:
+
+```js
+@Component({
+  template: `<div *ngFor="let item of list">{{item}}</div>`
+})
+export class ScrollComponent {  
+  list = Array.from({length: 100000}).map((_, i) => i);
+}
+```
+
+When the browser renders the app, it has to render 100,000 individual `<div>` elements. This might be fine for simple text nodes, but any complexity in the repeated template will not scale well, and any event listeners will be multiplied significantly.
+
+To add virtual scrolling and avoid those problems, you need to create a viewport by wrapping the list in a `<cdk-virtual-scroll-viewport>` element:
+
+```js/1-3
+@Component({
+  template: `<cdk-virtual-scroll-viewport>
+    <div *ngFor="let item of list">{{item}}</div>
+    </cdk-virtual-scroll-viewport>`
+})
+export class ScrollComponent {
+  list = Array.from({length: 100000}).map((_, i) => i);
+}
+```
+
+Because `ScrollingModule` dynamically renders subsets of the list, you have to specify the height of the viewport via standard CSS. You also need to give the viewport a hint about its content by specifying the `itemSize`. The module uses this information to determine how many items to keep in the DOM at a given time and how to render an appropriately sized scrollbar.
+
+
+```js/1
+@Component({
+  template: `<cdk-virtual-scroll-viewport itemSize="18" style="height:80vh">
+    <div *ngFor="let item of list">{{item}}</div>
+    </cdk-virtual-scroll-viewport>`
+})
+export class ScrollComponent {  
+  list = Array.from({length: 100000}).map((_, i) => i);
+}
+```
+
+Finally, convert `*ngFor` to `*cdkVirtualFor`:
+
+```js
+@Component({
+  template: `<cdk-virtual-scroll-viewport itemSize="18" style="height:80vh">
+    <div *cdkVirtualFor="let item of list">{{item}}</div>
+    </cdk-virtual-scroll-viewport>`
+})
+export class ScrollComponent {  
+  list = Array.from({length: 100000}).map((_, i) => i);
+}
+```
+
+ Instead of iterating through the entire list, the viewport will dynamically identify and iterate through the correct subset of the list for the user. Now when the user loads the page, the CDK should render the subset of the list that fits on the screen (plus a bit of buffer), and any scrolling events in the viewport will load and render the appropriate subset of the list:
+
+![A video showing the CDK rendering subsets of a list as the user scrolls](./render-subset.gif)
```

Comment:
@StephenFluin, can we use the original videos here instead of a gif, because of the better compression/quality?